src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -82,6 +82,12 @@ protected override async Task RemoveKeyAsync(string cacheKey)
             catch (Exception ex)
             {
                 _logger.LogError($"[IdWebCache] Connection issue encountered with Distributed cache. Currently using In Memory cache only. Error message: {ex.Message} ");
+
+                if (_distributedCacheOptions.OnL2CacheFailure != null && _distributedCacheOptions.OnL2CacheFailure(ex))
```

Comment:
Good pattern to apply everywhere. Try/catch can be refactored into a method/generic which takes an action

Reply:
fixed.

---
In reply to: [587808630](https://github.com/AzureAD/microsoft-identity-web/pull/1046#discussion_r587808630) [](ancestors = 587808630)